### DISCUSSION TITLE
Asynchronous Mutations

### DISCUSSION THREAD

**Discussion by bo55vxr** [ChilliCream/hotchocolate#5149] [Q&A]
I have a whole suite of APIs that I am in the process of fronting with Hot Chocolate. A majority of these APIs use asynchronous request-reply pattern (API returns 202 ACCEPT + location) so that the consuming application does not have to wait for the response, but can either implement polling, or come back later using the provided location. NB: Some of the APIs are long-running.

How can a consuming application maintain the same pattern where a mutation may take a long time (hours maybe) and then come back  to get the response.

**Comment by PascalSenn** [comment]
@bo55vxr 

You would just return an identifier and then poll over a query/mutation 
You can use Unions or interfaces to make the polling nicer. 

```graphql
{
   enqueueFoobar(input: $input) { 
      identifier
  }
}

{ 
    foobarStatus(identifier: $identifier) {
      ... on Running {
         message
      }
      ... on ErrorResult {
         errors {
            message
         }
     }
     ... on FooBarResult {
         foobar {
           result 
        }
    }
}
```
